packages/solid/store/src/store.ts

```diff
@@ -219,17 +231,14 @@ export function updatePath(current: StoreNode, path: any[], traversed: PropertyK
   } else setProperty(current, part, value);
 }
 
-export type DeepReadonly<T> = 0 extends 1 & T
-  ? T
-  : T extends NotWrappable
-  ? T
+export type DeepReadonly<T, W = any> =
+  T extends DeepMutable<W> ? W
+  : T extends NotWrappable ? T
```

Comment:
I am quite certain that this causes `T` to be `any` in the default case. There are type tests in `store/test/*.spec.ts`, check that those are fine. I'm not sure if this needed to be changed bar removing the extra `any` check.